Exiled.API/Features/Map.cs

```diff
@@ -60,14 +63,28 @@ public static ReadOnlyCollection<Room> Rooms
             {
                 if (RoomsValue.Count == 0)
                 {
-                    Room.ResetRoomIds();
-                    RoomsValue.AddRange(GameObject.FindGameObjectsWithTag("Room")
-                        .Select(r => new Room(r.name, r.transform, r.transform.position)));
-
-                    // Add the surface as a room.
-                    const string surfaceRoomName = "Root_*&*Outside Cams";
-                    var surfaceTransform = GameObject.Find(surfaceRoomName).transform;
-                    RoomsValue.Add(new Room(surfaceRoomName, surfaceTransform, surfaceTransform.position));
+                    List<Transform> roomTransforms = ListPool<Transform>.Shared.Rent();
+
+                    // Search for SECTR_Sector instead of tag. It is faster and includes Pocket dimension.
+                    roomTransforms.AddRange(
+                        Object.FindObjectsOfType<SECTR_Sector>()
+                            .Select(sector => sector.transform));
+
+                    // If no rooms were found, it means a plugin is trying to access this before the map is created.
+                    if (roomTransforms.Count == 0)
+                    {
+                        ListPool<Transform>.Shared.Return(roomTransforms);
+                        throw new NullReferenceException("Plugin is trying to access Rooms before they are created.");
```

Comment:
```suggestion
                        throw new InvalidOperationException("Plugin is trying to access Rooms before they are created.");
```

Because that isn't really a null ref

Reply:
Sounds good!